### PR TITLE
Rewrite 'which workflow' page

### PR DESCRIPTION
**Pull request by paulbalaji**
#### Description

Rewrite 'which workflow' page + other minor changes

#### Tests

improbadoc serve

#### Documentation

this is the damn documentation

### CODE REVIEW

File: docs/modules/core-and-feature-module-overview.md [hunk 1]
```diff
@@ -1,10 +1,7 @@
-[//]: # (Doc of docs reference 11)
-[//]: # (TODO - which module is ECS or GO-MB specific and which is generic?)
-
 <%(TOC)%>
 
 # Feature Modules Overview
-_This document relates to both [MonoBehaviour and ECS workflows]({{urlRoot}}/reference/workflows/which-workflow)._
+_This document relates to both [MonoBehaviour and ECS workflows]({{urlRoot}}/reference/workflows/overview)._
```
Comment (jamiebrynes7) [inline docs/modules/core-and-feature-module-overview.md]: Newline between header and text

File: docs/projects/fps/tutorial.md [hunk 2]
```diff
@@ -303,7 +304,7 @@ From these we can derive some rules about how the `UnityClient` and `UnityGameLo
 
 ---
 
-The FPS Starter Project uses the SpatialOS GDK's [MonoBehaviour workflow]({{urlRoot}}/reference/workflows/which-workflow). In this workflow SpatialOS entities are represented by Unity prefabs. Crucially, you can use different prefabs to represent the same type of entity on different types of workers. This allows you to separate client-side and server-side entity representation, as we planned above.
+The FPS Starter Project uses the SpatialOS GDK's [MonoBehaviour workflow]({{urlRoot}}/reference/workflows/overview). In this workflow SpatialOS entities are represented by Unity prefabs. Crucially, you can use different prefabs to represent the same type of entity on different types of workers. This allows you to separate client-side and server-side entity representation, as we planned above.
```
Comment (jamiebrynes7) [inline docs/projects/fps/tutorial.md]: Can link to the header directly?

File: docs/projects/fps/tutorial.md [hunk 3]
```diff
@@ -403,9 +404,9 @@ Here, we access the **current** data of the `HealthPickup` component of the unde
 
 No, exactly how entities are represented on each of your workers is up to you.
 
-The GDK also offers an [ECS workflow]({{urlRoot}}/reference/workflows/which-workflow) which represents them as a grouping of Unity ECS entity and components.
+The GDK also offers an [ECS workflow]({{urlRoot}}/reference/workflows/overview) which represents them as a grouping of Unity ECS entity and components.
 
-If you are more familiar with the traditional Unity GameObject style of development then the GDK provides a [MonoBehaviour workflow]({{urlRoot}}/reference/workflows/which-workflow) for you.
+If you are more familiar with the traditional Unity GameObject style of development then the GDK provides a [MonoBehaviour workflow]({{urlRoot}}/reference/workflows/overview) for you.
```
Comment (jamiebrynes7) [inline docs/projects/fps/tutorial.md]: Link to headers here too?

File: docs/reference/concepts/entity-lifecycle.md [hunk 4]
```diff
@@ -1,8 +1,7 @@
-[//]: # (Doc of docs reference 21)
-[//]: # (TODO - technical author pass)
 <%(TOC)%>
+
 # Entity lifecycle
-_This document relates to both [MonoBehaviour and ECS workflows]({{urlRoot}}/reference/workflows/which-workflow)._
+_This document relates to both [MonoBehaviour and ECS workflows]({{urlRoot}}/reference/workflows/overview)._
```
Comment (jamiebrynes7) [inline docs/reference/concepts/entity-lifecycle.md]: N e w l i n e

File: docs/reference/concepts/logging.md [hunk 5]
```diff
@@ -1,7 +1,7 @@
-[//]: # (TODO: Move this doc from the ecs folder)
 <%(TOC)%>
+
 #  Logs
- _This document relates to both the [MonoBehaviour and ECS workflows]({{urlRoot}}/reference/workflows/which-workflow)._
+ _This document relates to both the [MonoBehaviour and ECS workflows]({{urlRoot}}/reference/workflows/overview)._
```
Comment (jamiebrynes7) [inline docs/reference/concepts/logging.md]: NL

File: docs/reference/concepts/snapshots.md [hunk 6]
```diff
@@ -1,6 +1,7 @@
 <%(TOC)%>
+
 # Snapshots
-_This document relates to both [MonoBehaviour and ECS workflows]({{urlRoot}}/reference/workflows/which-workflow)._
+_This document relates to both [MonoBehaviour and ECS workflows]({{urlRoot}}/reference/workflows/overview)._
```
Comment (jamiebrynes7) [inline docs/reference/concepts/snapshots.md]: `\n`

File: docs/reference/concepts/worker.md [hunk 7]
```diff
@@ -1,8 +1,8 @@
-[//]: # (Doc of docs reference 15)
 <%(TOC)%>
+
 # Workers
 
-Before reading this document, make sure you are familiar with the [MonoBehaviour and  ECS workflow]({{urlRoot}}/reference/workflows/which-workflow).
+Before reading this document, make sure you are familiar with the [MonoBehaviour and  ECS workflow]({{urlRoot}}/reference/workflows/overview).
```
Comment (jamiebrynes7) [inline docs/reference/concepts/worker.md]: This suggestions that you need to be familiar with _both_ of these which isn't strictly true.

File: docs/reference/mobile/overview.md [hunk 8]
```diff
@@ -1,12 +1,13 @@
...
+
 # Mobile Support Overview
 
 <%(Callout type="warn" message=" Mobile support is in [pre-alpha](https://docs.improbable.io/reference/latest/shared/release-policy#maturity-stages). Please follow our [Roadmap](https://github.com/spatialos/gdk-for-unity/projects/1) to learn more about updates to this in future releases.")%>
 
 Before starting with mobile development, make sure you are familiar with
 
   * [The SpatialOS GDK for Unity]({{urlRoot}}/reference/overview)
-  * [The different workflows]({{urlRoot}}/reference/workflows/which-workflow)
+  * [The different workflows]({{urlRoot}}/reference/workflows/overview)
```
Comment (jamiebrynes7) [inline docs/reference/mobile/overview.md]: Tbh remove this line entirely? @jessicafalk ?

File: docs/reference/overview.md [hunk 9]
```diff
@@ -13,7 +14,7 @@ If you are new to SpatialOS, you can check out the SpatialOS concept documentati
 
 We recommend you take a look at the following three pages first:
 
-* The introduction to the two workflows: [the MonoBehaviour and ECS Workflows]({{urlRoot}}/reference/workflows/which-workflow.md). <br/>
+* The introduction to the two workflows: [the MonoBehaviour and ECS Workflows]({{urlRoot}}/reference/workflows/overview.md). <br/>
```
Comment (jamiebrynes7) [inline docs/reference/overview.md]: Can you drop the extension on the link to make it consistent 🙏

Comment (jamiebrynes7) [reply docs/reference/overview.md]: Also replace the <br/> with a `\n`

File: docs/reference/workflows/ecs/entity-contracts.md [hunk 10]
```diff
@@ -1,9 +1,7 @@
-[//]: # (Doc of docs reference 4)
-[//]: # (TODO - Tech writer pass)
-
 <%(TOC)%>
+
 # ECS entity contracts
-  _This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/which-workflow)._
+  _This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/overview)._
```
Comment (jamiebrynes7) [inline docs/reference/workflows/ecs/entity-contracts.md]: e n i l w e n

File: docs/reference/workflows/ecs/interaction/authority.md [hunk 11]
```diff
@@ -1,6 +1,7 @@
 <%(TOC)%>
+
 # ECS: Authority
- _This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/which-workflow.md)._
+ _This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/overview.md)._
```
Comment (jamiebrynes7) [inline docs/reference/workflows/ecs/interaction/authority.md]: `std::endl`

File: docs/reference/workflows/ecs/interaction/commands/component-commands.md [hunk 12]
```diff
@@ -1,15 +1,14 @@
-[//]: # (Doc of docs reference 34)
-[//]: # (Tech writer review)
 <%(TOC)%>
+
 # ECS: Component commands
- _This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/which-workflow)._
+ _This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/overview)._
```
Comment (jamiebrynes7) [inline docs/reference/workflows/ecs/interaction/commands/component-commands.md]: `System.Environment.NewLine`

File: docs/reference/workflows/ecs/interaction/commands/component-commands.md [hunk 13]
```diff
@@ -1,15 +1,14 @@
...
 
 Before reading this document, make sure you are familiar with:
 
 * [Read and write access]({{urlRoot}}/reference/glossary#authority)
 
 ## About commands
-Commands are SpatialOS's equivalent of [remote procedure calls (Wikipedia)](https://en.wikipedia.org/wiki/Remote_procedure_call). You use commands to send messages between two [workers]({{urlRoot}}/reference/concepts/worker). Commands are relevant to both [MonoBehaviour and ECS workflows]({{urlRoot}}/reference/workflows/which-workflow).<br/>
+Commands are SpatialOS's equivalent of [remote procedure calls (Wikipedia)](https://en.wikipedia.org/wiki/Remote_procedure_call). You use commands to send messages between two [workers]({{urlRoot}}/reference/concepts/worker). Commands are relevant to both [MonoBehaviour and ECS workflows]({{urlRoot}}/reference/workflows/overview).<br/>
```
Comment (jamiebrynes7) [inline docs/reference/workflows/ecs/interaction/commands/component-commands.md]: `os.linesp`

File: docs/reference/workflows/ecs/interaction/component-updates.md [hunk 14]
```diff
@@ -1,9 +1,7 @@
-[//]: # (Doc of docs reference 33)
-[//]: # (TODO - Tech writer review)
-
 <%(TOC)%>
+
 #  ECS: Component updates
- _This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/which-workflow)._
+ _This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/overview)._
```
Comment (jamiebrynes7) [inline docs/reference/workflows/ecs/interaction/component-updates.md]: `Console.WriteLine("");`

File: docs/reference/workflows/ecs/interaction/events.md [hunk 15]
```diff
@@ -1,6 +1,7 @@
 <%(TOC)%>
+
 # ECS: Events
- _This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/which-workflow)._
+ _This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/overview)._
```
Comment (jamiebrynes7) [inline docs/reference/workflows/ecs/interaction/events.md]: ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ ‌‌ 

^ one of those

File: docs/reference/workflows/ecs/reactive-components.md [hunk 16]
```diff
@@ -1,6 +1,7 @@
 <%(TOC)%>
+
 # Reactive components
- _This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/which-workflow)._
+ _This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/overview)._
```
Comment (jamiebrynes7) [inline docs/reference/workflows/ecs/reactive-components.md]: `\r\n`

File: docs/reference/workflows/ecs/system-update-order.md [hunk 17]
```diff
@@ -1,10 +1,7 @@
-[//]: # (Doc of docs reference 31.1)
-[//]: # (TODO - Tech writer review)
-[//]: # (TODO - use discussions about content in here https://docs.google.com/document/d/1MPTP1qEo9LaYxFGLQFEN2SqEzu9MxlKjVfOYKPUbTXg/edit)
-
 <%(TOC)%>
+
 # System update order
- _This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/which-workflow)._
+ _This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/overview)._
```
Comment (jamiebrynes7) [inline docs/reference/workflows/ecs/system-update-order.md]: `\r` on old Macs

File: docs/reference/workflows/ecs/temporary-components.md [hunk 18]
```diff
@@ -1,12 +1,7 @@
...
-[//]: # (TODO - Tech writer review)
-[//]: # (Editorial review status: Engineer review only)
-[//]: # (Questions to deal with: need to describe what a tick is. Needs to link to a doc on system execution order. Needs to link to docs describing the existing components which use this)
-[//]: # (1. Editorial review - JIRA TICKET: UTY-628)
-
 <%(TOC)%>
+
 # Temporary components
- _This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/which-workflow)._
+ _This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/overview)._
```
Comment (jamiebrynes7) [inline docs/reference/workflows/ecs/temporary-components.md]: `U+000A`

File: docs/reference/workflows/ecs/worker-entity.md [hunk 19]
```diff
@@ -1,9 +1,7 @@
-[//]: # (Doc of docs reference 15.2a)
-[//]: # (TODO: Move to the ecs folder)
-
 <%(TOC)%>
+
 # Worker entity
-_This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/which-workflow)._
+_This document relates to the [ECS workflow]({{urlRoot}}/reference/workflows/overview)._
```
Comment (jamiebrynes7) [inline docs/reference/workflows/ecs/worker-entity.md]: `0x0a`

File: docs/reference/workflows/monobehaviour/creating-workers.md [hunk 20]
```diff
@@ -1,6 +1,7 @@
 <%(TOC)%>
+
 # Creating workers
-_This document relates to the [MonoBehaviour workflow]({{urlRoot}}/reference/workflows/which-workflow)._
+_This document relates to the [MonoBehaviour workflow]({{urlRoot}}/reference/workflows/overview)._
```
Comment (jamiebrynes7) [inline docs/reference/workflows/monobehaviour/creating-workers.md]: `sys.props("line.separator")`

File: docs/reference/workflows/monobehaviour/interaction/commands/create-delete-spatialos-entities.md [hunk 21]
```diff
@@ -1,9 +1,7 @@
-[//]: # (Doc of docs reference 26)
-[//]: # (TODO - technical author pass)
-
 <%(TOC)%>
+
 # How to create and delete entities
-_This document relates to the [MonoBehaviour workflow]({{urlRoot}}/reference/workflows/which-workflow)._
+_This document relates to the [MonoBehaviour workflow]({{urlRoot}}/reference/workflows/overview)._
```
Comment (jamiebrynes7) [inline docs/reference/workflows/monobehaviour/interaction/commands/create-delete-spatialos-entities.md]: `System.lineSeparator()`

File: docs/reference/workflows/monobehaviour/interaction/commands/world-commands.md [hunk 22]
```diff
@@ -1,19 +1,15 @@
-[//]: # (Doc of docs reference 9)
-[//]: # (TODO - Tech writer pass)
-[//]: # (TODO - explain what “handling the response based on the information contained in this object” means - see note below.)
-[//]: # (TODO - link to status codes for error messages - see note below.)
-
 <%(TOC)%>
+
 # World commands
-_This document relates to the [MonoBehaviour workflow]({{urlRoot}}/reference/workflows/which-workflow#spatialos-entities)._
+_This document relates to the [MonoBehaviour workflow]({{urlRoot}}/reference/workflows/overview#monobehaviour-centric-workflow)._
```
Comment (jamiebrynes7) [inline docs/reference/workflows/monobehaviour/interaction/commands/world-commands.md]: `require('os').EOL`

File: docs/reference/workflows/monobehaviour/interaction/reader-writers/component-data-updates.md [hunk 23]
```diff
@@ -1,10 +1,7 @@
-[//]: # (Doc of docs reference 6.2)
-[//]: # (TODO - Tech writer pass)
-[//]: # (TODO - Remove “> Currently updating any field of the component will trigger the callback for all <component property name>Updated`.” line if if this PR gets merged in: https://github.com/spatialos/UnityGDK/pull/438 )
-
 <%(TOC)%>
+
 # Readers and Writers: Component data and updates
-_This document relates to the [MonoBehaviour workflow]({{urlRoot}}/reference/workflows/which-workflow#spatialos-entities)._
+_This document relates to the [MonoBehaviour workflow]({{urlRoot}}/reference/workflows/overview#monobehaviour-centric-workflow)._
```
Comment (jamiebrynes7) [inline docs/reference/workflows/monobehaviour/interaction/reader-writers/component-data-updates.md]: `<br/>`

File: docs/reference/workflows/monobehaviour/interaction/commands/world-commands.md [hunk 24]
```diff
@@ -1,19 +1,15 @@
...
 
 Before reading this document, make sure you are familiar with
 
   * [How to interact with SpatialOS using MonoBehaviours]({{urlRoot}}/reference/workflows/monobehaviour/interaction/reader-writers/lifecycle)
   * [SpatialOS entities: Creating entity templates]({{urlRoot}}/reference/concepts/entity-templates)
 
 ## About commands
-Commands are SpatialOS's equivalent of [remote procedure calls (Wikipedia)](https://en.wikipedia.org/wiki/Remote_procedure_call). You use commands to send messages between two [workers]({{urlRoot}}/reference/concepts/worker). Commands are relevant to both [MonoBehaviour and ECS workflows]({{urlRoot}}/reference/workflows/which-workflow).<br/>
+Commands are SpatialOS's equivalent of [remote procedure calls (Wikipedia)](https://en.wikipedia.org/wiki/Remote_procedure_call). You use commands to send messages between two [workers]({{urlRoot}}/reference/concepts/worker). Commands are relevant to both [MonoBehaviour and ECS workflows]({{urlRoot}}/reference/workflows/overview).<br/>
```
Comment (jamiebrynes7) [inline docs/reference/workflows/monobehaviour/interaction/commands/world-commands.md]: What is "new line"?

File: docs/reference/workflows/monobehaviour/interaction/reader-writers/index.md [hunk 25]
```diff
@@ -1,10 +1,7 @@
-[//]: # (Doc of docs reference 6.1)
-[//]: # (TODO - Tech writer pass)
-[//]: # (TODO - See if `option` struct needs defining)
-
 <%(TOC)%>
+
 # Readers and Writers
-_This document relates to the [MonoBehaviour workflow]({{urlRoot}}/reference/workflows/which-workflow#spatialos-entities)._
+_This document relates to the [MonoBehaviour workflow]({{urlRoot}}/reference/workflows/overview#monobehaviour-centric-workflow)._
```
Comment (jamiebrynes7) [inline docs/reference/workflows/monobehaviour/interaction/reader-writers/index.md]:
```
if (!new_line->does_exist()) 
{
	console.write(get_way_of_saying_newline());
}
```

File: docs/reference/workflows/monobehaviour/interaction/reader-writers/lifecycle.md [hunk 26]
```diff
@@ -1,8 +1,7 @@
-[//]: # (Doc of docs reference 5.2)
-[//]: # (TODO - tech writer review)
 <%(TOC)%>
+
 # Readers and Writers: Lifecycle
-_This document relates to the [MonoBehaviour workflow]({{urlRoot}}/reference/workflows/which-workflow)._
+_This document relates to the [MonoBehaviour workflow]({{urlRoot}}/reference/workflows/overview)._
```
Comment (jamiebrynes7) [inline docs/reference/workflows/monobehaviour/interaction/reader-writers/lifecycle.md]: `NSLineSeparatorCharacter`

File: docs/reference/workflows/overview.md [hunk 27]
```diff
@@ -0,0 +1,29 @@
+# Overview
+
+The SpatialOS GDK for Unity provides functionality to develop games in both of Unity's paradigms:
+
+* Object-oriented: MonoBehaviours and GameObjects
+* Data-oriented: Entity-Component-System (ECS)
+
+You can stick to familiar MonoBehaviour-based development, check out the new ECS workflow or try a combination of both.
```
Comment (jamiebrynes7) [inline docs/reference/workflows/overview.md]: Oxford comma 😉

File: docs/reference/workflows/overview.md [hunk 28]
```diff
@@ -0,0 +1,29 @@
...
+
+You can stick to familiar MonoBehaviour-based development, check out the new ECS workflow, or try a combination of both.
+
+> However you choose to develop your game, it is important to understand what [SpatialOS entities]({{urlRoot}}/reference/glossary#spatialos-entity) are and how they relate to each workflow.
+
+## MonoBehaviour-centric workflow
+
+In the GDK, SpatialOS entities can be linked to GameObjects. [Readers and Writers]({{urlRoot}}/reference/workflows/monobehaviour/interaction/reader-writers/index) allow you to inspect and change the state of components on SpatialOS entities using MonoBehaviours that you add to their linked GameObjects.
+
+You may find that not all SpatialOS entities need to be represented as a GameObjects. The [Game Object Creation Feature Module]({{urlRoot}}/modules/game-object-creation/overview) allows you to customise the GameObject creation and linking process.
```
Comment (jessicafalk) [inline docs/reference/workflows/overview.md]:
```suggestion
You may find that not all SpatialOS entities need to be represented as GameObjects. The [Game Object Creation Feature Module]({{urlRoot}}/modules/game-object-creation/overview) allows you to customise the GameObject creation and linking process.
```

I would also mention the feature module in the first sentence where you say that entities can be linked to GameObjects to make it clear that you need to use that module to have them linked

File: docs/reference/workflows/overview.md [hunk 29]
```diff
@@ -0,0 +1,29 @@
...
+
+## ECS-centric workflow
+
+ECS is a new Unity development paradigm, introduced by Unity in early 2018.
+
+In the GDK, each SpatialOS entity checked out by a worker has a corresponding ECS entity. To inspect or change the state of components on SpatialOS entities, you must read or write to the components on corresponding ECS entities.
+
+To learn more about ECS, check out:
```
Comment (jessicafalk) [inline docs/reference/workflows/overview.md]: not for this PR, but we really need some kind of introduction page on ECS. We don't have a single link here on how to get started with our ECS-centric workflow

Comment (paulbalaji) [reply docs/reference/workflows/overview.md]: I agree. Part of the plan is to have a similar overview page for MBs/ECS which then lays the ground work to link to worker entities / reader/writers / components / reactive vs nonreactive etc

File: docs/reference/workflows/overview.md [hunk 30]
```diff
@@ -0,0 +1,29 @@
...
+
+## MonoBehaviour-centric workflow
+
+In the GDK, SpatialOS entities can be linked to GameObjects. [Readers and Writers]({{urlRoot}}/reference/workflows/monobehaviour/interaction/reader-writers/index) allow you to inspect and change the state of components on SpatialOS entities using MonoBehaviours that you add to their linked GameObjects.
+
+You may find that not all SpatialOS entities need to be represented as a GameObjects. The [Game Object Creation Feature Module]({{urlRoot}}/modules/game-object-creation/overview) allows you to customise the GameObject creation and linking process.
+
+## ECS-centric workflow
+
+ECS is a new Unity development paradigm, introduced by Unity in early 2018.
```
[the remark this answers is not in the document]
Comment (paulbalaji) [reply docs/reference/workflows/overview.md]: how about

`ECS is an architectural pattern adopted by Unity in early 2018 to bring data-oriented design to the Unity game engine.`

**Pull request merged**